test/Microsoft.AspNetCore.Mvc.Razor.Test/RazorViewEngineTest.cs

```diff
@@ -1420,6 +1420,36 @@ public void GetAbsolutePath_ReturnsPagePathUnchanged_IfNotAPath(string executing
         }
 
         [Theory]
+        [InlineData("/Views/Home/Index.cshtml", "../Shared/_Partial.cshtml")]
+        [InlineData("/Views/Home/Index.cshtml", "..\\Shared\\_Partial.cshtml")]
+        [InlineData("/Areas/MyArea/Views/Home/Index.cshtml", "../../../../Views/Shared/_Partial.cshtml")]
+        [InlineData("/Views/Accounts/Users.cshtml", "../Test/../Shared/_Partial.cshtml")]
+        public void GetAbsolutePath_ResolvesPathTraversals(string executingFilePath, string pagePath)
+        {
+            // Arrange
+            var viewEngine = CreateViewEngine();
+
+            // Act
+            var result = viewEngine.GetAbsolutePath(executingFilePath, pagePath);
+
+            // Assert
+            Assert.Equal("/Views/Shared/_Partial.cshtml", result);
+        }
+
+        [Fact]
+        public void GetAbsolutePath_DoesNotResolvePathIfTraversalsEscapeTheRoot()
+        {
+            // Arrange
+            var viewEngine = CreateViewEngine();
+
+            // Act
+            var result = viewEngine.GetAbsolutePath("/Index.cshtml", "../Shared/_Layout.cshtml");
```

Comment:
Can you add another case where the `..` comes later, e.g. `Foo/../Bar/../../Oops.cshtml`?